src/main/java/uk/gov/pay/api/model/search/card/SearchRefunds.java

```diff
@@ -0,0 +1,131 @@
+
+        this.client = client;
+        this.objectMapper = objectMapper;
+        this.uriGenerator = uriGenerator;
+        this.baseUrl = configuration.getBaseUrl();
+    }
+
+    public Response getSearchResponse(Account account, Map<String, String> queryParams) {
+        if (account.getPaymentType().equals(TokenPaymentType.DIRECT_DEBIT)) {
+            throw new BadRequestException(PaymentError
+                    .aPaymentError(PaymentError.Code.SEARCH_PAYMENTS_VALIDATION_ERROR, "agreement_id"));
+        }
+
+        String url = uriGenerator.refundsURIWithParams(account, queryParams);
+        Response connectorResponse = client
+                .target(url)
+                .request()
+                .header(HttpHeaders.ACCEPT, APPLICATION_JSON)
+                .get();
+        logger.info("response from connector for refunds search: " + connectorResponse);
+        if (connectorResponse.getStatus() == SC_OK) {
+            return processResponse(account.getAccountId(), connectorResponse);
+        }
+        throw new SearchRefundsException(connectorResponse);
+    }
+
+    private Response processResponse(String accountId, Response connectorResponse) {
+        try {
+            JsonNode responseJson = connectorResponse.readEntity(JsonNode.class);
+            TypeReference<SearchRefundsResponse> typeRef = new TypeReference<SearchRefundsResponse>() {
+            };
+            SearchRefundsResponse searchResponse = objectMapper.readValue(responseJson.traverse(), typeRef);
+            List<RefundForSearchRefundsResult> results = searchResponse.getRefunds()
+                    .stream()
+                    .map(refund -> RefundForSearchRefundsResult.valueOf(refund))
+                    .collect(Collectors.toList());
+
+            HalRepresentation.HalRepresentationBuilder halRepresentation = HalRepresentation
+                    .builder()
+                    .addProperty("results", results);
+            return Response.ok().entity(decoratePagination(Long.valueOf(accountId), halRepresentation, searchResponse)
+                    .build().toString())
+                    .build();
+        } catch (IOException | ProcessingException ex) {
+            throw new SearchRefundsException(ex);
+        }
+    }
+
+    protected HalRepresentation.HalRepresentationBuilder decoratePagination(Long accountId,
+                                                                            HalRepresentation.HalRepresentationBuilder halRepresentationBuilder,
+                                                                            SearchRefundsResponse pagination) {
+        try {
+            halRepresentationBuilder
+                    .addProperty("count", pagination.getCount())
+                    .addProperty("total", pagination.getTotal())
+                    .addProperty("page", pagination.getPage());
+            addLink(halRepresentationBuilder, "self", transformIntoPublicUri(
```

Comment:
i see there's already logic for pagination for search payments, if it's the same we should definitely move that up the hierarchy and make both payments and refund use it - we don't want to have pagination logic in two places.

The difference I can see is that you removed first, last, prev and next pages from here - is this by design? What happens if we have multiple pages?

Reply:
I think we will add this soon as it would mean changing connector and the pact files.